drivers/java/src/main/java/com/rethinkdb/RethinkDB.java

```diff
@@ -8,9 +9,16 @@
     /**
      * The Singleton to use to begin interacting with RethinkDB Driver
      */
-    public static final RethinkDB r = new RethinkDB();
-
+    private static ObjectMapper mapper = null;
     public Connection.Builder connection() {
         return Connection.build();
     }
+
+    public static void setObjectMapper(ObjectMapper mapper1){
+        mapper = mapper1;
+    }
+    public static ObjectMapper getObjectMapper(){
+        if (mapper != null) return mapper;
+        return new ObjectMapper();
```

Comment:
If we're going to create a new mapper, we should probably assign it to `mapper` before returning it.